pkg/svcat/service-catalog/instance.go

```diff
@@ -236,7 +258,7 @@ func (sdk *SDK) WaitForInstance(ns, name string, interval time.Duration, timeout
 		func() (bool, error) {
 			instance, err = sdk.RetrieveInstance(ns, name)
 			if nil != err {
-				if apierrors.IsNotFound(err) {
+				if apierrors.IsNotFound(errors.Cause(err)) {
```

Comment:
Seems to me that with this fix right here (using errors.Cause), that we can remove the new function that you introduced `WaitForInstanceToNotExist`?

Otherwise if you want to keep the separate methods, then this logic should change (because if we are waiting for an operation that shouldn't delete resource, then this is actually an error case, when previously it was how I handled exiting cleanly when a resource was deleted.

Reply:
I'm actually seeing an immediate return from svcat delete instance --wait as it is.  I believe svcat was hitting the WaitForInstance() and was seeing it in a ready state with no op yet in progress and returning.   I'd prefer to use the WaitForInstanceToNotExist() - - I'll remove the NotFound  conditional as you suggested.